dotnet/src/webdriver/SessionId.cs

```diff
@@ -17,29 +17,53 @@
 // under the License.
 // </copyright>
 
+using System;
+using System.Diagnostics.CodeAnalysis;
+
+#nullable enable
+
 namespace OpenQA.Selenium
 {
     /// <summary>
     /// Provides a mechanism for maintaining a session for a test
     /// </summary>
-    public class SessionId
+    public class SessionId : IEquatable<SessionId>, IEquatable<string>
     {
-        private string sessionOpaqueKey;
+        private readonly string? sessionOpaqueKey;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="SessionId"/> class
         /// </summary>
         /// <param name="opaqueKey">Key for the session in use</param>
-        public SessionId(string opaqueKey)
+        [Obsolete("Call SessionId.Create")]
+        public SessionId(string? opaqueKey)
```

Comment:
I believe `opaqueKey` cannot be `null`, otherwise I don't see a sense for `SessionId` as a class.

Reply:
Yes I completely agree, that's why I made this constructor obsolete. Currently, users are capable of setting this value to `null` with no errors. We may do the same in `WebDriver`, because we could be doing `new SessionId(null).ToString()`, which returns `null`.

I don't want to introduce any breaking changes here, which is why I introduced the `Create` method below.

Reply:
Introducing new API just to avoid a breaking change of the class leads to mess. Just break, it is not a big deal.

Reply:
Done, made sure to avoid nullability issues in `WebDriver` which may have been using this

Reply:
Thx, not sure if we should even mention about it. But it is better to mention, so renaming PR to mention it.